### PR TITLE
Faster machine, more workers

### PR DESCRIPTION
**Pull request by porter-stripe**
## Summary
- More workers
- Faster machine
- Anywhere from 10-15% reduction in run times for UI tests, marginal increase in credit usage.

## Motivation
- CI sprint

## Testing
- Manual via CI

## Changelog
N/A

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -583,14 +583,14 @@ workflows:
         - scheme: PaymentSheet Example
         - test_plan: PaymentSheet Example-Shard2
         - log_formatter: xcbeautify
-        - xcodebuild_options: -maximum-concurrent-test-simulator-destinations 2 -maximum-parallel-testing-workers 2
+        - xcodebuild_options: -maximum-concurrent-test-simulator-destinations 4 -maximum-parallel-testing-workers 4
```
Comment (porter-stripe) [inline bitrise.yml:586]: Putting this at 6 workers causes it to fail for some reason

Comment (davidme-stripe) [reply bitrise.yml:586]: We should be consistent between shards IMO. It gets flaky when we add too many, and I don't want engineers writing tests to have to micromanage the shard count

File: bitrise.yml [hunk 2]
```diff
@@ -565,14 +565,14 @@ workflows:
...
-        - xcodebuild_options: -maximum-concurrent-test-simulator-destinations 2 -maximum-parallel-testing-workers 2
+        - xcodebuild_options: -maximum-concurrent-test-simulator-destinations 6 -maximum-parallel-testing-workers 6
     - deploy-to-bitrise-io@2: {}
     before_run:
     - prep_all
     meta:
       bitrise.io:
         stack: osx-xcode-15.0.x
-        machine_type_id: g2-m1.8core
+        machine_type_id: g2-m1-max.5core
```
Comment (davidme-stripe) [inline bitrise.yml:575]: Seems good!

File: bitrise.yml (hunk 1 above)
Comment (porter-stripe) [reply bitrise.yml]: Turned out to be an actual flake not caused by more workers, it's now fixed. Both shards are now using 6 workers. https://stripe.slack.com/archives/C02HQBW38JX/p1721675483160809

**Review by davidme-stripe — APPROVED** [review]

**Pull request merged**